test/servinge2e/servicemesh_test.go

```diff
@@ -32,41 +46,67 @@ func TestKsvcWithServiceMeshSidecar(t *testing.T) {
 
 	tests := []struct {
 		name string
-		annotations map[string]string
-		expectIstioSidecar bool
+		labels map[string]string      // Ksvc labels
+		annotations map[string]string // Revision template annotations
+		useProxy bool                 // Whether to use an httpproxy image to connect to the ksvc URL (for cluster-local ksvcs)
+		expectIstioSidecar bool       // Whether it is expected for the istio-proxy sidecar to be injected into the pod
 	}{{
 		// Requests go via gateway -> activator -> pod , by default
 		// Verifies the activator can connect to the pod
 		name: "sidecar-via-activator",
 		annotations: map[string]string{
-			"sidecar.istio.io/inject":          "true",
+			"sidecar.istio.io/inject":	"true",
 		},
 		expectIstioSidecar: true,
 	},{
 		// Requests go via gateway -> pod ( activator should be skipped if burst capacity is disabled and there is at least 1 replica)
 		// Verifies the gateway can connect to the pod directly
 		name: "sidecar-without-activator",
 		annotations: map[string]string{
-			"sidecar.istio.io/inject":											"true",
-			"autoscaling.knative.dev/targetBurstCapacity": 	"0",
-			"autoscaling.knative.dev/minScale": 						"1",
+			"sidecar.istio.io/inject":                      "true",
+			"autoscaling.knative.dev/targetBurstCapacity":  "0",
+			"autoscaling.knative.dev/minScale":             "1",
 		},
 		expectIstioSidecar: true,
 	},{
 		// Verifies the "sidecar.istio.io/inject" annotation is really what decides the istio-proxy presence
 		name: "no-sidecar",
 		annotations: map[string]string{
-			"sidecar.istio.io/inject":					"false",
+			"sidecar.istio.io/inject": "false",
 		},
 		expectIstioSidecar: false,
+	},{
+		// A cluster-local variant of the "sidecar-via-activator" scenario
+		name: "local-sidecar-via-activator",
+		labels: map[string]string{
+			"serving.knative.dev/visibility": "cluster-local",
```

Comment:
You can use https://github.com/knative/serving/blob/master/pkg/apis/serving/register.go#L97. (It will move to networking repo in the future, though.)